src/main/java/org/jitsi/videobridge/BitrateController.java

```diff
@@ -0,0 +1,613 @@
+     * represented by their IDs. Required for backwards compatibility with
+     * existing LastN code.
+     */
+    private List<String> forwardedEndpoints = INITIAL_EMPTY_LIST;
+
+    /**
+     * Initializes a new {@link BitrateController} instance which is to
+     * belong to a particular {@link VideoChannel}.
+     *
+     * @param dest the {@link VideoChannel} that owns this instance.
+     */
+    BitrateController(VideoChannel dest)
+    {
+        this.dest = dest;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PacketTransformer getRTPTransformer()
+    {
+        return rtpTransformer;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PacketTransformer getRTCPTransformer()
+    {
+        return rtcpTransformer;
+    }
+
+    /**
+     * Defines a packet filter that controls which packets to be written into
+     * the {@link Channel} that owns this {@link BitrateController}.
+     *
+     * @param data true if the specified packet/<tt>buffer</tt> is RTP, false if
+     * it is RTCP.
+     * @param buf the <tt>byte</tt> array that holds the packet.
+     * @param off the offset in <tt>buffer</tt> at which the actual data begins.
+     * @param len the number of <tt>byte</tt>s in <tt>buffer</tt> which
+     * constitute the actual data.
+     * @return <tt>true</tt> to allow the specified packet/<tt>buffer</tt> to be
+     * written into the {@link Channel} that owns this {@link BitrateController}
+     * ; otherwise, <tt>false</tt>
+     */
+    public boolean accept(boolean data, byte[] buf, int off, int len)
+    {
+        long ssrc;
+        if (data)
+        {
+
+            ssrc = RawPacket.getSSRCAsLong(buf, off, len);
+            if (ssrc < 0)
+            {
+                return false;
+            }
+        }
+        else
+        {
+            ssrc = RTCPHeaderUtils.getSenderSSRC(buf, off, len);
+            if (ssrc < 0)
+            {
+                return false;
+            }
+
+        }
+
+        SimulcastController simulcastController
+            = ssrcToBitrateController.get((int) ssrc);
+
+        return simulcastController != null
+            && simulcastController.rtpTranslatorWillWrite(data, buf, off, len);
+    }
+
+    /**
+     * Computes a new bitrate allocation for every endpoint in the conference,
+     * and updates the state of this instance so that bitrate allocation is
+     * eventually met.
+     *
+     * @param conferenceEndpoints the ordered list of {@link Endpoint}s
+     * participating in the multipoint conference with the dominant (speaker)
+     * {@link Endpoint} at the beginning of the list i.e. the dominant speaker
+     * history. This parameter is optional but it can be used for performaance;
+     * if it's omitted it will be fetched from the
+     * {@link ConferenceSpeechActivity}.
+     */
+    public void update(List<Endpoint> conferenceEndpoints)
```

Comment:
I think we need to protect the iterate/write access to ssrcToBitrateController, since multiple threads could execute update()